src/dcp/dcp.c

```diff
@@ -86,18 +86,26 @@ static void daos_set_paths(
         }
 
         /* figure out if prefix is on dst or src for 
-         * for copying container subsets */
-        char* src_ret = strstr(src_path, dfs_prefix);
-        if (src_ret != NULL) {
-            mfu_src_file->type = DAOS;
-            uuid_copy(src_pool_uuid, dattr.da_puuid);
-            uuid_copy(src_cont_uuid, dattr.da_cuuid);
-            argpaths[0] = src_path + strlen(dfs_prefix);
-        } else {
-            mfu_dst_file->type = DAOS;
-            uuid_copy(dst_pool_uuid, dattr.da_puuid);
-            uuid_copy(dst_cont_uuid, dattr.da_cuuid);
-            argpaths[1] = dst_path + strlen(dfs_prefix);
+         * copying container subsets */
+	int prefix_len = strlen(dfs_prefix);
+	if (strncmp(src_path, dfs_prefix, prefix_len) == 0) {
+            /* if we have another character, it must be '/' */
+            if (strlen(src_path) > prefix_len &&
```

Comment:
Due to this nested if structure, I'm not sure this will do what you want for something like:
```
dfs_prefix=/dfs
src=/dfs_not
dst=/dfs
```
Maybe move this nested if to a function and then call that from daos_set_paths?

Reply:
@adammoody hmm yes in that case it doesn't check the dest path. I added a check_prefix function that I think covers all of the cases now hopefully 👍